purchase_requisition_line_description/models/product.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.model
+    def _name_search(
+        self, name, args=None, operator='ilike', limit=100, name_get_uid=None
+    ):
+        if name and self.env.context.get('vendor_data_search'):
+            positive_operators = ['=', 'ilike', '=ilike', 'like', '=like']
+            product_ids = []
+            if operator in positive_operators:
+                product_ids = self._search(
+                    [('seller_ids.product_code', '=', name)] + args,
+                    limit=limit,
+                    access_rights_uid=name_get_uid,
+                )
+                if not product_ids:
+                    product_ids = self._search(
+                        [('seller_ids.product_name', '=', name)] + args,
+                        limit=limit,
+                        access_rights_uid=name_get_uid,
+                    )
+        else:
+            product_ids = self._search(
+                args, limit=limit, access_rights_uid=name_get_uid
+            )
+        return self.browse(product_ids).name_get()
```

Comment:
```suggestion
        if name and self.env.context.get('vendor_data_search'):
            product_ids = self._search(
                    [('name', operator, name)] + args,
                    limit=limit,
                    access_rights_uid=name_get_uid,
                )
            positive_operators = ['=', 'ilike', '=ilike', 'like', '=like']
            if operator in positive_operators:
                product_ids += self._search(
                    [('seller_ids.product_code', operator, name)] + args,
                    limit=limit,
                    access_rights_uid=name_get_uid,
                )
                if not product_ids:
                    # this condition will never run
                    product_ids += self._search(
                        [('seller_ids.product_name', operator, name)] + args,
                        limit=limit,
                        access_rights_uid=name_get_uid,
                    )
           return self.browse(product_ids).name_get()
        return super()._name_search(name, args, operator, limit, name_get_uid)
```

Reply:
@i-vyshnevska  this code doesn't work. Why do we need to return name_get here?

Reply:
so far this works:
```

        product_ids = []
        if name and self.env.context.get('vendor_data_search'):
            product_ids = self._search(
                    [('name', operator, name)] + args,
                    limit=limit,
                    access_rights_uid=name_get_uid,
                )
            positive_operators = ['=', 'ilike', '=ilike', 'like', '=like']
            if operator in positive_operators:
                product_ids += self._search(
                    ['|',('seller_ids.product_code', operator, name),('seller_ids.product_name', operator, name)] + args,
                    limit=limit,
                    access_rights_uid=name_get_uid,
                )
        return self.browse(product_ids).name_get()
```
pls clarify what you plan to achieve with name_get and name_search instead of browsing on record, how it is done in default method.